core/src/components/modal/test/dark-mode/model.e2e.ts

```diff
@@ -0,0 +1,29 @@
+import { expect } from '@playwright/test';
+import { test } from '@utils/test/playwright';
+
+test.describe('modal: dark mode', () => {
+  test('should render the correct text color when outside ion-content', async ({ page, skip }) => {
+    test.info().annotations.push({
+      type: 'issue',
+      description: 'https://github.com/ionic-team/ionic-framework/issues/26060',
+    });
+
+    skip.rtl();
+
+    await page.goto('/src/components/modal/test/dark-mode');
+
+    const ionModalDidPresent = await page.spyOnEvent('ionModalDidPresent');
+
+    const button = page.locator('#basic-modal');
+    await button.click();
+
+    await ionModalDidPresent.next();
+
+    const modal = await page.locator('ion-modal');
+    await expect(modal).toHaveClass(/show-modal/);
```

Comment:
```suggestion
```

`ionModalDidPresent` fires when the modal is visible and has completed any transition, so this check should not be needed.